src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -674,14 +637,15 @@ bool BaseCouplingScheme::accelerate()
     getAcceleration()->performAcceleration(getAccelerationData());
   }
 
-  // Store data for conv. measurement, acceleration, or extrapolation
-  storeData();
-
   // extrapolate new input data for the solver evaluation in time.
   if (convergence && (_extrapolationOrder > 0)) {
-    extrapolateData(getAccelerationData());
+    storeWindowData();
```

Comment:
Why `storeWindowData` only when extrapolation is used? We will need it for more later and should already test it now moving forward.

Reply:
Agree. We will have to change a few things in the initialization of the Waveform (by default initialize with `_extrapolationOrder = 1`), since otherwise there will not be any place to store to. But this is not performance critical and generalizes the code.

Reply:
I implemented this change here: 6e516d7. We can even move `storeWindowData` further up, but then we will probably leave `BaseCouplingScheme::doImplicitStep`. I think it also makes sense to always support extrapolation / waveform relaxation (not only for implicit coupling schemes). We can also do an extrapolation with an explicit coupling scheme as long as the data is there.

Reply:
I just realized we cannot move it up too far, since we can only store the data after the convergence check: 4ea5d13

Reply:
We could store and potentially overwrite it in every iteration. I would argue that we should not worry about performance already now.

Reply:
44f05b2 is a quite big change in this direction. Idea: Store data of current window in 0th column and old data in additional columns. When a window is converged, we call a special method that moves the data down one column and does extrapolation. Zero order extrapolation is the same like no extrapolation and just uses the last value as an initial guess. Higher order extrapolation is also possible.

I guess we could move `updateThisWindow()` further up, if we want. What we now definitely need is more tests for the coupling scheme.